resources/default_pmd_rulesets.xml

```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0"?>
+<ruleset name="Custom ruleset"
+        xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+    <description>
+        This ruleset checks special qualities suggested by the SAP S4HANA Cloud SDK
+    </description>
+
+    <rule name="HandleExceptionsRule"
+            message="Please pass the exception to a logger or error handling method or reference it in a new throw exception."
+            class="com.sap.cloud.sdk.quality.pmd.rules.HandleExceptionsRule">
+        <description>
+            In order to allow post mortem analysability log the exception in the catch block or in a called handling method or reference it in a new thrown exception.
+        </description>
+        <priority>1</priority>
+
+        <example>
+            <![CDATA[
+    protected List<CostCenterDetails> convertResult(Result result) {
+        try {
+            return result.asList(CostCenterDetails.class);
+        }
+        catch( final Exception e) {
+            doNothing();
+        }
+    }
+]]>
+        </example>
+    </rule>
+
+    <rule name="ReferenceExceptionRule"
+            message="If you log an error in an exception you should reference it."
+            class="com.sap.cloud.sdk.quality.pmd.rules.ReferenceExceptionRule">
```

Comment:
As the rule is implemented as part of the cloud sdk, I think it somehow needs to be added to the project, right?

Reply:
With the recent changes the cloud sdk's pmd plugin is used.